pkg/database/connection.go

```diff
@@ -97,3 +102,81 @@ func checkError(err error, logMessage string) {
 		klog.Error(logMessage, " ", err)
 	}
 }
+
+func (dao *DAO) UpsertCluster(resource model.Resource) {
```

Comment:
Can we move these new functions to a different file?

Reply:
Moved